.cursor/rules/accessibility-requirements.mdc

```diff
@@ -0,0 +1,271 @@
+
+## Guidelines
+
+### Do This ✅
+
+```typescript
+// Good: Provide meaningful screen reader content
+import { VisuallyHidden } from '@/components/Common'
+
+const DataTable = ({ data }) => (
+  <table>
+    <thead>
+      <tr>
+        <th>
+          Employee Name
+          <VisuallyHidden>Sort by employee name</VisuallyHidden>
+        </th>
+        <th>
+          Pay Amount
+          <VisuallyHidden>Sort by pay amount</VisuallyHidden>
+        </th>
+        <th>
+          <VisuallyHidden>Actions for employee</VisuallyHidden>
+        </th>
+      </tr>
+    </thead>
+    <tbody>
+      {data.map(employee => (
+        <tr key={employee.id}>
+          <td>{employee.name}</td>
+          <td>{employee.amount}</td>
+          <td>
+            <Button aria-label={`Edit ${employee.name}`}>
+              <Icon name="edit" />
+              <VisuallyHidden>Edit employee</VisuallyHidden>
+            </Button>
+          </td>
+        </tr>
+      ))}
+    </tbody>
+  </table>
+)
+
+// Good: Meaningful button labels
+const PayrollActions = ({ payroll }) => (
+  <div>
+    <Button
+      aria-label={`Submit payroll for ${payroll.payPeriod}`}
+      onClick={() => submitPayroll(payroll.id)}
+    >
+      <Icon name="submit" />
+      <VisuallyHidden>Submit payroll</VisuallyHidden>
+    </Button>
+
+    <Button
+      aria-label={`Cancel payroll for ${payroll.payPeriod}`}
+      onClick={() => cancelPayroll(payroll.id)}
+    >
+      <Icon name="cancel" />
+      <VisuallyHidden>Cancel payroll</VisuallyHidden>
+    </Button>
+  </div>
+)
+```
+
+### Avoid This ❌
+
+```typescript
+// Bad: Empty strings for screen readers
+const DataTable = ({ data }) => (
+  <table>
+    <thead>
+      <tr>
+        <th>Employee Name</th>
+        <th>Pay Amount</th>
+        <th></th> {/* Empty header - no context for screen readers */}
+      </tr>
+    </thead>
+    <tbody>
+      {data.map(employee => (
+        <tr key={employee.id}>
+          <td>{employee.name}</td>
+          <td>{employee.amount}</td>
+          <td>
+            <Button> {/* No label - screen reader doesn't know purpose */}
+              <Icon name="edit" />
+            </Button>
+          </td>
+        </tr>
+      ))}
+    </tbody>
+  </table>
+)
+
+// Bad: No meaningful labels for icon buttons
+const PayrollActions = ({ payroll }) => (
+  <div>
+    <Button onClick={() => submitPayroll(payroll.id)}>
+      <Icon name="submit" />
+    </Button>
+
+    <Button title="Cancel"> {/* Title is not read by all screen readers */}
+      <Icon name="cancel" />
+    </Button>
+  </div>
+)
+```
+
+## Key Accessibility Patterns
+
+### Screen Reader Content
+
+#### Use VisuallyHidden for:
+
+- **Icon-only buttons**: Provide text description
+- **Table headers**: Explain column purpose
+- **Form labels**: When visual label isn't sufficient
+- **Status updates**: Announce state changes
```

Comment:
Would visually hidden help with this? This seems potentially more involved with appending text to live regions?